Sources/Logger/Loggers/SystemLogger.swift

```diff
@@ -6,26 +6,32 @@
 //
 
 import Foundation
-import os
+import OSLog
 
 /// Pre-built logger that wraps system os_logger
 public class SystemLogger: Logging {
     private let logEntryEncoder: LogEntryEncoding
-    private var logger: OSLog
+    private let prefix: String
+    private let systemLogger: (OSLogType, String) -> Void
 
     public init(
         subsystem: String,
         category: String,
-        logEntryEncoder: LogEntryEncoding = LogEntryEncoder()
+        logEntryEncoder: LogEntryEncoding = LogEntryEncoder(),
+        prefix: String = "",
+        systemLogger: ((OSLogType, String) -> Void)? = nil
```

Comment:
Nedava větší smysl nahradit
`systemLogger: ((OSLogType, String) -> Void)? = nil`
za
`systemLogger: @escaping ((OSLogType, String) -> Void) = SystemLogger.defaultSystemLogger `
note: Je potřeba vytvorit statickou funkci  `defaultSystemLogger `

Protoze takhle se naprosto zbytečně vytváří instance OSLog, která se nevyužije (pokud systemLogger není nil). A není primocare co dela ten optional.

Reply:
Nemyslim si, ze vytvareni `OSLog` nekoho boli ale jinak souhlasim, prepisme to...